src/libraries/Microsoft.Extensions.Http/src/DefaultHttpMessageHandlerBuilder.cs

```diff
@@ -42,6 +45,18 @@ public override HttpMessageHandler Build()
                 throw new InvalidOperationException(message);
             }
 
+#if NET8_0_OR_GREATER
+            // The MeterFactory property is available on handlers in .NET 8 or later.
+            if (PrimaryHandler is HttpClientHandler httpClientHandler)
+            {
+                httpClientHandler.MeterFactory = _meterFactory;
```

Comment:
I updated it to use a filter.

Defaults or configure don't work because the meter factory should be applied last. If someone changes the handler, such as to a SocketsHttpHandler, then we want to apply the meter factory after that change.